Sources/Target/Common/ThreadBase.cpp

```diff
@@ -30,5 +30,17 @@ ErrorCode ThreadBase::modifyRegisters(
   action(state);
   return writeCPUState(state);
 }
+
+ErrorCode ThreadBase::beforeResume() {
+  if (!_process->isAlive())
+    return kErrorProcessNotFound;
+
+  BreakpointManager *bpm = _process->hardwareBreakpointManager();
+  if (bpm != nullptr) {
+    bpm->enable();
```

Comment:
A function to enable breakpoints on a specific thread currently doesn't exist in the `BreakpointManager` API, since software breakpoints don't care about threads. I could add it, but given that thread-specific hardware breakpoints aren't a thing in lldb, I don't think it's necessary, and just complicates the API.